### PR TITLE
Add explainer for WritableStreamDefaultController's AbortSignal

### PR DESCRIPTION
**Pull request by nidhijaju**
Add writable-stream-abort-signal-explainer.md.

### CODE REVIEW

File: writable-stream-abort-signal-explainer.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+
+```javascript
+const ws = new WritableStream({
+  write(controller) {
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, 1000);
+      controller.signal.addEventListener('abort',
+        () => reject(controller.abortReason()));
+});
```
Comment (ricea) [inline writable-stream-abort-signal-explainer.md:49]: I think we're missing two '}' here.

**Review by ricea — COMMENTED** [review]
lgtm, but let's wait and see if @domenic or @MattiasBuelens have any comments.

File: writable-stream-abort-signal-explainer.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+writer.write('some data');
+await writer.abort();
+```
+
+This example shows a use case of this feature with WebTransport.
+
+```javascript
+const wt = new WebTransport(...);
+await wt.ready;
+const ws = await wt.createUnidirectiaonalStream();
```
Comment (MattiasBuelens) [inline writable-stream-abort-signal-explainer.md:85]:
```suggestion
const ws = await wt.createUnidirectionalStream();
```

File: writable-stream-abort-signal-explainer.md (hunk 1 above)
Comment (nidhijaju) [reply writable-stream-abort-signal-explainer.md:51]: Done, I've added the correct parentheses/brackets here.

**Review by MattiasBuelens — COMMENTED** [review]
Looks good, just one small nit. 🙂

File: writable-stream-abort-signal-explainer.md (hunk 2 above)
Comment (nidhijaju) [reply writable-stream-abort-signal-explainer.md]: Done :)

**Review by MattiasBuelens — APPROVED** [review]

File: writable-stream-abort-signal-explainer.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+# WritableStream controller AbortSignal Explainer
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:1]:
```suggestion
# `WritableStream` controller `AbortSignal` Explainer
```

File: writable-stream-abort-signal-explainer.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+On [WritableStreamDefaultController](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller)
+(the controller argument that is passed to underlying sinks):
+
+*   [abortReason](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-abortreason)
+    *   The argument passed to `writable.abort()` or `writer.abort()`. Undefined if no argument was passed or `abort()`
+    hasn't been called.
+*   [signal](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-signal)
+    *   An AbortSignal. By using `signal.addEventListener('abort', …)` an underlying sink can abort the pending write
+    or close operation when the stream is aborted.
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:29]:
```suggestion
*   [`abortReason`](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-abortreason): The argument passed to `writable.abort()` or `writer.abort()`. Undefined if no argument was passed or `abort()`
    hasn't been called.
*   [`signal`](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-signal): An `AbortSignal`. By using `signal.addEventListener('abort', …)` an underlying sink can abort the pending write
    or close operation when the stream is aborted.
```

File: writable-stream-abort-signal-explainer.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+The WritableStream API does not change. Instead, the existing `abort()` operation will now signal abort.
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:31]:
```suggestion
The `WritableStream` API does not change. Instead, the existing `abort()` operation will now signal abort.
```

File: writable-stream-abort-signal-explainer.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+
+## Examples
+
+These are some examples of Javascript which can be used for writable streams once this is implemented:
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:36]:
```suggestion
These are some examples of JavaScript which can be used for writable streams once this is implemented:
```

File: writable-stream-abort-signal-explainer.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+called it stops immediately.
+
+
+```javascript
+const ws = new WritableStream({
+  write(controller) {
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, 1000);
+      controller.signal.addEventListener('abort',
+        () => reject(controller.abortReason()));
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:48]:
```suggestion
        () => reject(controller.abortReason));
```

File: writable-stream-abort-signal-explainer.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  }
+});
+const writer = ws.getWriter();
+
+writer.write(99);
+await writer.abort();
+```
+
+
+This example shows integration with an existing API that uses AbortSignal. In this case, each write() triggers a POST
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:59]:
```suggestion
This example shows integration with an existing API that uses `AbortSignal`. In this case, each `write()` triggers a POST
```

File: writable-stream-abort-signal-explainer.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+await writer.abort();
+```
+
+
+
+## Goals
+
+*   Allow writes to be aborted more quickly and efficiently.
+*   WebTransport will be able to use WritableStreamDefaultController.signal to make
+[SendStream's write](https://w3c.github.io/webtransport/#sendstream-write) and
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:101]:
```suggestion
[`SendStream`'s `write()`](https://w3c.github.io/webtransport/#sendstream-write) and
```

File: writable-stream-abort-signal-explainer.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+[close](https://w3c.github.io/webtransport/#sendstream-close) abortable.
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:102]:
```suggestion
[`close()`](https://w3c.github.io/webtransport/#sendstream-close) abortable.
```

File: writable-stream-abort-signal-explainer.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+## Non-Goals
+
+*   Exposing a method to abort individual operations without aborting the stream as a whole. The semantics of this
+would be unclear and confusing.
+
+
+## User Benefits
+
+*   Allows `abort` operation to complete more quickly, which avoids wasted resources for sites that take advantage of it.
```
Comment (domenic) [inline writable-stream-abort-signal-explainer.md:113]:
```suggestion
*   Allows the abort operation to complete more quickly, which avoids wasted resources for sites that take advantage of it.
```

**Review by domenic — APPROVED** [review]
LGTM with mostly typographical nits. One substantial issue in the example code, `abortReason()` should be `abortReason`.

**Comment by nidhijaju** [comment]
Thank you for reviewing, everyone! :) Please double check, and help me merge if all looks good.

**Pull request merged**